sa/saro.go

```diff
@@ -994,6 +994,11 @@ func (ssa *SQLStorageAuthorityRO) GetValidOrderAuthorizations2(ctx context.Conte
 		return nil, errIncompleteRequest
 	}
 
+	qualifiedAuthzFields := make([]string, 0)
+	for _, col := range strings.Split(authzFields, " ") {
+		qualifiedAuthzFields = append(qualifiedAuthzFields, fmt.Sprintf("authz2.%s", col))
+	}
```

Comment:
Efficiency nit; the only ambiguous field is `id`. I believe you can avoid a couple allocations and a few loops in this codepath with the following:

```suggestion
	qualifiedAuthzFields := strings.Split(authzFields, " ")
	for i, col := range qualifiedAuthzFields {
		if col == "id," {
			qualifiedAuthzFields[i] = "authz2.id,"
			break
		}
	}
```

You should consider adding a small snippet above this logic to indicate the issue that this addresses.